core/src/scene/sceneLoader.cpp

```diff
@@ -64,6 +65,97 @@ void printFilters(const SceneLayer& layer, int indent){
     }
 };
 
+void SceneLoader::applyGlobalProperties(Node& node, Scene& scene) {
+    switch(node.Type()) {
+    case NodeType::Scalar:
+        {
+            std::string key = node.Scalar();
+            const std::string wildcard = "_$_";
+            if (key.compare(0, 7, "global.") == 0) {
+                key.replace(0, 7, "");
+                key = std::regex_replace(key, std::regex("\\."), wildcard);
+                if (scene.m_globals.find(key) != scene.m_globals.end()) {
+                    auto& value = scene.m_globals[key];
+                    switch (value.which()) {
```

Comment:
This seems like a risky thing to switch on - if we add another type to the variant then all of these cases could break with no compile-time error. I think you can check the type with `is<Type>()` or something, but this becomes redundant if the map of globals just stores yaml nodes.